utils/obsproc/Smap2Ioda.h

```diff
@@ -0,0 +1,104 @@
+    explicit Smap2Ioda(const eckit::Configuration & fullConfig, const eckit::mpi::Comm & comm)
+      : NetCDFToIodaConverter(fullConfig, comm) {
+      variable_ = "Salinity";
+    }
+
+    // Read netcdf file and populate iodaVars
+    gdasapp::IodaVars providerToIodaVars(const std::string fileName) final {
+      oops::Log::info() << "Processing files provided by SMAP" << std::endl;
+
+      // Open the NetCDF file in read-only mode
+      netCDF::NcFile ncFile(fileName, netCDF::NcFile::read);
+
+      // Get number of obs
+      int dim0  = ncFile.getDim("phony_dim_0").getSize();
+      int dim1  = ncFile.getDim("phony_dim_1").getSize();
+      int nobs = dim0 * dim1;
+
+      // Set the int metadata names
+      // TODO(AFE) fill this in: intMetadataNames = {"pass", "cycle", "mission"};
+      std::vector<std::string> intMetadataNames = {};
+
+      // Set the float metadata name
+      // TODO(AFE) fill this in:  floatMetadataNames = {"mdt"};
+      std::vector<std::string> floatMetadataNames = {};
+      // Create instance of iodaVars object
+      gdasapp::IodaVars iodaVars(nobs, floatMetadataNames, intMetadataNames);
+
+      // TODO(AFE): these arrays can be done as 1D vectors, but those need proper ushorts in
+      // the input files, at odd with the current ctests
+      float lat[dim0][dim1];  // NOLINT
+      ncFile.getVar("lat").getVar(lat);
+
+      float lon[dim0][dim1];  // NOLINT
+      ncFile.getVar("lon").getVar(lon);
+
+      float sss[dim0][dim1];  // NOLINT
+      ncFile.getVar("smap_sss").getVar(sss);
+
+      float sss_error[dim0][dim1];  // NOLINT
+      ncFile.getVar("smap_sss_uncertainty").getVar(sss_error);
+
+      unsigned short sss_qc[dim0][dim1];  // NOLINT
+      ncFile.getVar("quality_flag").getVar(sss_qc);
+
+      // "UTC seconds of day"
+      float obsTime[dim1];  // NOLINT
+      ncFile.getVar("row_time").getVar(obsTime);
+
+      int startYear;
+      netCDF::NcGroupAtt attributeStartYear = ncFile.getAtt("REV_START_YEAR");
+      attributeStartYear.getValues(&startYear);
+
+      int startDay;
+      netCDF::NcGroupAtt attributeStartDay = ncFile.getAtt("REV_START_DAY_OF_YEAR");
+      attributeStartDay.getValues(&startDay);
+
+      // calculate the seconds of Jan 1 of startyear since unix epoch
+      std::tm tm{};
+      // defaults are zero, Jan is zero
+      tm.tm_year = startYear - 1900;
+      tm.tm_mday = 1;
+      time_t unixStartYear = mktime(&tm);
+
+      int unixStartDay = unixStartYear + startDay * 86400;
+
+      int loc;
+      for (int i = 0; i < dim0; i++) {
+        for (int j = 0; j < dim1; j++) {
+          loc = i * dim1 + j;
+          iodaVars.longitude(loc) = lon[i][j];
+          iodaVars.latitude(loc) = lat[i][j];
+          iodaVars.obsVal(loc) = sss[i][j];
+          iodaVars.obsError(loc) = sss_error[i][j];
+          iodaVars.preQc(loc) = sss_qc[i][j];
+          iodaVars.datetime(loc) =  static_cast<int64_t>(obsTime[j] + unixStartDay);
```

Comment:
I'm uneasy with time manipulation ... Did you check that the output is correct?

Reply:
Yes - the dates in the output (including from the original datafile) landed within the time specified elsewhere in the file

Reply:
Specifically, I checked the times in a unix epoch converter